ARE/components/actuator.irtrans/src/main/java/eu/asterics/component/actuator/irtrans/IrTransInstance.java

```diff
@@ -294,43 +318,43 @@ public void start() {
      */
     @Override
     public void stop() {
-        closeSocket();
+        closeTcpSocket(tcpSocket);
+        closeTcpSocket(tcpSocketRead);
```

Comment:
close tcpSocketRead in the reading thread, because here you can't be sure that the reference variable already holds the reference to the socket although it was already opened. This is due to thread caching. To prevent caching you could declare tcpSocketRead as

private volatile Socket tcpSocketRead

nevertheless it's better to close it in the reader thread.